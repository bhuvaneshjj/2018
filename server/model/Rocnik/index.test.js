'use strict';

const db = require('../../db');
const Kategorie = require('../Kategorie.js');
const Rocnik = require('./index.js');

beforeAll(async () => {
  await db.dropDatabase();
});

afterAll(async () => {
  await db.disconnect();
});

it('vytvoř ročník s jednou kategorií', async () => {
  const kategorie = new Kategorie({
    typ: 'maraton',
    pohlavi: 'zena',
    minVek: 40,
    maxVek: 49,
    maStartCislo: true
  });
  await kategorie.save();

  const rocnik = new Rocnik({ datum: '2017-06-10' });
  rocnik.kategorie.push({
    typ: 'maraton',
    kategorie: [kategorie.id],
    startCisla: '1-100',
    startovnePredem: 150,
    startvneNaMiste: 200
  });
  await rocnik.save();

  const rocniky = await Rocnik.find({}, { _id: 0 }).populate('kategorie.kategorie', { _id: 0 });
  expect(rocniky).toMatchSnapshot();

  await Rocnik.remove({});
});

it('vytvoř ročník s ubytováním', async () => {
  const kategorie1 = new Kategorie({
    typ: 'maraton',
    pohlavi: 'zena',
    minVek: 40,
    maxVek: 49,
    maStartCislo: true
  });
  await kategorie1.save();
  const kategorie2 = new Kategorie({
    typ: 'maraton',
    pohlavi: 'zena',
    minVek: 50,
    maxVek: 59,
    maStartCislo: true
  });
  await kategorie2.save();

  const rocnik = new Rocnik({ datum: '2017-06-10' });
  rocnik.kategorie.push({
    typ: 'maraton',
    kategorie: [kategorie1.id, kategorie2.id],
    startCisla: '1-100',
    startovnePredem: 150,
    startvneNaMiste: 200
  });
  rocnik.ubytovani.push({ den: 'pátek', poplatek: 50 });
  rocnik.ubytovani.push({ den: 'sobota', poplatek: 60 });
  await rocnik.save();

  const rocniky = await Rocnik.find({}, { _id: 0 }).populate('kategorie.kategorie', { _id: 0 });
  expect(rocniky).toMatchSnapshot();

  await Rocnik.remove({});
});
