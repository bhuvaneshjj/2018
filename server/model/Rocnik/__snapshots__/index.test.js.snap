// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vytvoř ročník s jednou kategorií 1`] = `
Array [
  Object {
    "__v": 0,
    "datum": 2017-06-10T00:00:00.000Z,
    "kategorie": Array [
      Object {
        "kategorie": Array [
          Object {
            "__v": 0,
            "maStartCislo": true,
            "maxVek": 49,
            "minVek": 40,
            "pohlavi": "zena",
            "typ": "maraton",
          },
        ],
        "startCisla": "1-100",
        "startovnePredem": 150,
        "startvneNaMiste": 200,
        "typ": "maraton",
      },
    ],
    "ubytovani": Array [],
  },
]
`;

exports[`vytvoř ročník s ubytováním 1`] = `
Array [
  Object {
    "__v": 0,
    "datum": 2017-06-10T00:00:00.000Z,
    "kategorie": Array [
      Object {
        "kategorie": Array [
          Object {
            "__v": 0,
            "maStartCislo": true,
            "maxVek": 49,
            "minVek": 40,
            "pohlavi": "zena",
            "typ": "maraton",
          },
          Object {
            "__v": 0,
            "maStartCislo": true,
            "maxVek": 59,
            "minVek": 50,
            "pohlavi": "zena",
            "typ": "maraton",
          },
        ],
        "startCisla": "1-100",
        "startovnePredem": 150,
        "startvneNaMiste": 200,
        "typ": "maraton",
      },
    ],
    "ubytovani": Array [
      Object {
        "den": "pátek",
        "poplatek": 50,
      },
      Object {
        "den": "sobota",
        "poplatek": 60,
      },
    ],
  },
]
`;
